src/Chapter2.hs

```diff
@@ -857,7 +896,8 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind = foldl' (\xs x -> x : xs) []
```

Comment:
Extremely nice work on using standard functions! If you want to write it even shorter, you can use a standard function `flip` and apply it to the cons operator `:`. And you will get an idiomatic implementation of `reverse`

Reply:
oh! I was wishing I had that function, I didn't realize it was a real function. So cool!